### ISSUE TITLE
Visualization: backend changes on scatterplot viz

### ISSUE THREAD

**Issue by moontrip** [VEuPathDB/web-eda#134]
Danielle said there would be two backend changes early next week on scatterplot viz: a) smoothedMean option would be removed from scatter; b) including errors passed along from R for smoothed means. 

Will adopt them at scatterplot viz. For a), scatterplot controls would need to be changed too

**Comment by moontrip** [comment]
I could check the response from feature site and realized that there exists a new response, SmoothedMeanError. 

Probably this needs to be handled correctly (discussing with Danielle)

**Comment by moontrip** [comment]
Hi @dmfalke @bobular, 
I have examined and tested the new response with smoothedMeanError and I want to hear your opinion about this.

TL;DR
_For Gates demo, we will be fine without further change as it would be a scripted demo, however, in my opinion, we should bypass io-ts error by using union type for some response parameters to avoid a backend issue_

For detailed information:
Such a error case occurs when overlay variable is selected with a specific combination of X/Y axis variable. One example that causes an error is: X = Sleeping rooms in dwelling; Y = Persons living in house; Overlay = Country (see below screenshot as well). The smoothedMeanError response is comprised of a string, which, if not empty, contains R error message: e.g., `Error in smooth.construct.cr.smooth.spec(object, data, knots) : x has insufficient unique values to support 10 knots: reduce k.` - as it stated, the case does not have enough unique x data to compute smoothed mean (see below screenshot for reference). 

One issue in the backend change is that some response types are incorrect when smoothedMeanError is not empty (i.e., with message): smoothedMeanY and smoothedMeanSE should return number array but currently return string array, which causes io-ts error. Thus I reported this to Danielle by [making an issue at plot.data](https://github.com/VEuPathDB/plot.data/issues/40). So the current feature site shows the io-ts error with the above example and plot option of `Smoothed Mean With Raw`. Certainly, this issue can be temporarily bypassed by allowing both number and string arrays for those parameters (at data-api.ts).

I have also tested how the current scatter plot viz responds with the bypass (using union type) and actually it works without any further changes. The error cases, India and Kenya in the example case, are plotted without error, but their smoothed mean and confidence interval are basically non-existent because smoothedMeanX/Y/SE response are empty for error case: yes it makes sense anyway. A screenshot is attached for the case.

![scatter-error1](https://user-images.githubusercontent.com/12802305/121544996-1fe2ac80-c9d8-11eb-8e28-9d438fa5032d.png)

**Comment by dmfalke** [comment]
Thanks for the detailed information, DK. We should definitely not do
anything related to showing errors at this point. I think we need a more
focused discussion on this topic, at least involving Danielle and Steve,
but possibly others as well.

You mentioned an io-ts error on the feature site. I think it’s okay for now
since the demo will be scripted, so it can be avoided.
